kong/tools/dns.lua

```diff
@@ -37,6 +37,30 @@ local setup_client = function(conf)
     noSynchronisation = conf.dns_no_sync,
   }
 
+  -- new dns client
+  if conf.legacy_dns_client == "off" then
```

Comment:
```suggestion
  if not conf.legacy_dns_client then
```

Reply:
We should use `not conf.legacy_dns_client`, don't compare to `false`

Reply:
fixed good catch!!

Reply:
here is a problem, for kong tools, like `bin/kong start`, there is no `conf` argument when it was called firstly.

So for `bin/kong` command, there is no good way to detect whether to use new dns client, so here we need to use the checking stmt in `resty/dns.lua`

```
-- Use the new dns client library instead. If you want to switch to the original
-- one, you can set `legacy_dns_client = on` in kong.conf.
if ngx.shared.kong_dns_cache and not _G.busted_legacy_dns_client then
  package.loaded["kong.dns.client"] = nil
  return require("kong.dns.client")
end
```